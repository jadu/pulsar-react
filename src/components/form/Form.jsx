import React, { Children } from 'react';

export default class Form extends React.Component {

  static defaultProps = {
    className: 'form',
    encType: 'application/x-www-form-urlencoded',
    method: 'POST'
  };

  render() {
    let {
      children,
      method,
      nonce,
      ...props
    } = this.props;

    return (
      <form {...props}>
        {method == 'PUT' &&
          <input name="_method" value="PUT" type="hidden" />
        }
        {nonce &&
          <input name="nonce" value={nonce} type="hidden" />
        }
        {children}
      </form>
    );
  }
}