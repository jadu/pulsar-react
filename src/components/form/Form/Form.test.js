'use strict';

import React from 'react';
import { render } from 'enzyme';

import Form from './Form';

it('Renders the form wrapper', () => {
  const tree = render(
    <Form>foo</Form>
  );
  expect(tree).toMatchSnapshot();
});

it('allows the PUT method', () => {
  const tree = render(
    <Form method="PUT">foo</Form>
  );
  expect(tree).toMatchSnapshot();
});

it('Allows a nonce to be supplied', () => {
  const tree = render(
    <Form nonce="baz">foo</Form>
  );
  expect(tree).toMatchSnapshot();
});
