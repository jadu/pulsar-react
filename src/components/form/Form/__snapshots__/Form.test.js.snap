// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Allows a nonce to be supplied 1`] = `
<form
  class="form"
  enctype="application/x-www-form-urlencoded"
>
  <input
    name="nonce"
    type="hidden"
    value="baz"
  />
  foo
</form>
`;

exports[`Renders the form wrapper 1`] = `
<form
  class="form"
  enctype="application/x-www-form-urlencoded"
>
  foo
</form>
`;

exports[`allows the PUT method 1`] = `
<form
  class="form"
  enctype="application/x-www-form-urlencoded"
>
  <input
    name="_method"
    type="hidden"
    value="PUT"
  />
  foo
</form>
`;
